diff --git a/widget/ContentCache.cpp b/widget/ContentCache.cpp
index 348e3b6718bd..3e30cfff906e 100644
--- a/widget/ContentCache.cpp
+++ b/widget/ContentCache.cpp
@@ -1248,6 +1248,28 @@ bool ContentCacheInParent::GetUnionTextRects(
   return !aUnionTextRect.IsEmpty();
 }
 
+bool ContentCacheInParent::GetCachedCaretRect(
+    uint32_t aOffset, LayoutDeviceIntRect& aCaretRect) const {
+  if (!mSelection || !mSelection->mHasRange) {
+    return false;
+  }
+
+  if (!mSelection->IsCollapsed() || mSelection->StartOffset() != aOffset) {
+    return false;
+  }
+
+  if (!mCaret || !mCaret->HasRect() || mCaret->Offset() != aOffset) {
+    return false;
+  }
+
+  if (PendingEventsNeedingAck() || HasPendingCommit()) {
+    return false;
+  }
+
+  aCaretRect = mCaret->mRect;
+  return true;
+}
+
 bool ContentCacheInParent::GetCaretRect(uint32_t aOffset,
                                         bool aRoundToExistingOffset,
                                         LayoutDeviceIntRect& aCaretRect) const {
