diff --git a/widget/ContentCache.h b/widget/ContentCache.h
index 1143a1de9bbe..d4f28f14c768 100644
--- a/widget/ContentCache.h
+++ b/widget/ContentCache.h
@@ -409,6 +409,9 @@ class ContentCacheInParent final : public ContentCache {
   bool HandleQueryContentEvent(WidgetQueryContentEvent& aEvent,
                                nsIWidget* aWidget) const;
 
+  bool GetCachedCaretRect(uint32_t aOffset,
+                          LayoutDeviceIntRect& aCaretRect) const;
+
   /**
    * OnCompositionEvent() should be called before sending composition string.
    * This returns true if the event should be sent.  Otherwise, false.
